fpm/src/fpm_model.f90

```diff
@@ -0,0 +1,68 @@
+module fpm_model
+
+! Definition and validation of the backend model
+
+use fpm_command_line, only: fpm_build_settings
+use fpm_filesystem, only: exists, join_path
+use fpm_manifest, only: package_t, default_library, default_executable
+use fpm_manifest_executable, only: executable_t
+use fpm_sources, only: resolve_module_dependencies, add_sources_from_dir, &
+                       add_executable_sources, srcfile_t
+use fpm_strings, only: string_t
```

Comment:
None of these should be imported here, except `string_t` if we use it in the model derived types (we do not seem to currently).